lib/ansible/modules/storage/vexata/vexata_pg.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Sandeep Kasargod (sandeep@vexata.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+module: vexata_pg
+version_added: 2.9
+short_description: Manage port groups on Vexata VX100 storage arrays
+description:
+    - Create, deletes or modify port groups on a Vexata VX100 array.
+author:
+  - Sandeep Kasargod (@vexata)
+options:
+  name:
+    description:
+      - Port group name.
+    required: true
+    type: str
+  state:
+    description:
+    - Creates/Modifies port group when present or delete when absent.
+    - Port groups that are in one or more export groups cannot be deleted
+      without first deleting those export groups.
+    default: present
+    choices: [ present, absent ]
+    type: str
+  ports:
+    description:
+    - List of array port ids, the port id range is 0-15.
+    type: list
+extends_documentation_fragment:
+    - vexata.vx100
+'''
+
+EXAMPLES = r'''
+- name: Create port group named pg1 with four ports.
+  vexata_pg:
+    name: pg1
+    ports:
+    - 0
+    - 1
+    - 8
+    - 9
+    state: present
+    array: vx100_ultra.test.com
+    user: admin
+    password: secret
+
+- name: Modify existing port group named pg1 to have 6 ports.
+  vexata_pg:
+    name: pg1
+    ports:
+    - 0
+    - 1
+    - 2
+    - 8
+    - 9
+    - 10
+    state: present
+    array: vx100_ultra.test.com
+    user: admin
+    password: secret
+
+- name: Remove port group named pg1
+  vexata_pg:
+    name: pg1
+    state: absent
+    array: vx100_ultra.test.com
+    user: admin
+    password: secret
+'''
+
+RETURN = r'''
```

Comment:
Nice to have return values here for better UI/UX

Reply:
Rest LGTM

Reply:
Currently the module doesn't return anything (the facts module does). Can enhance subsequently.